### PR TITLE
refactor tests in nodemetric package

### LINKED ISSUE #172: [proposal] Unify the style of test code
**What is your proposal**:
Unify the style of test code:
https://github.com/koordinator-sh/koordinator/blob/main/pkg/slo-controller/nodemetric/nodemetric_controller_test.go

**Why is this needed**:
Consistent coding style so participants don't get confused.

**Is there a suggested solution, if so, please add it**:
refactor nodemetric tests

### PR DESCRIPTION
**Pull request by hormes**
Signed-off-by: Fansong Zeng <fanster.z@gmail.com>

fix #172

### CODE REVIEW

File: hack/run-test.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (hormes) [reply hack/run-test.sh:1]: to run test easily as the name, the detailed is described in usage function

File: pkg/slo-controller/nodemetric/nodemetric_controller_test.go [hunk 2]
```diff
@@ -19,161 +19,154 @@ package nodemetric
...
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/event"
 
 	slov1alpha1 "github.com/koordinator-sh/koordinator/apis/slo/v1alpha1"
 	"github.com/koordinator-sh/koordinator/pkg/slo-controller/config"
+	"github.com/koordinator-sh/koordinator/pkg/slo-controller/noderesource"
 )
 
-var _ = Describe("NodeMetric Controller", func() {
```
[the remark this answers is not in the document]
Comment (hormes) [reply pkg/slo-controller/nodemetric/nodemetric_controller_test.go:38]: case are equivalent, there is a PR behind the one-sided to refactor the code, then more tests will be added

**Review by jasonliu747 — COMMENTED** [review]
/lgtm

**Pull request merged**